### DISCUSSION TITLE
inject header Access-Control-Allow-Origin: * in WMR server

### DISCUSSION THREAD

**Discussion by wilberforce** [preactjs/wmr#889] [Q&A] [answered]
Hi,
I'm getting:
```
Access to script at 'http://localhost/webserver-v2.js' from origin 'http://aroha-garage-mini.local' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```
I  can't find any documentation on what to set in `wmr.config.mjs`, or does this need  a custom plugin?

Thanks

**Comment by marvinhagemeister** [comment, marked as answer]
It's currently not well documented in the docs, but WMR does allow adding custom middlewares to the server:

```js
// wmr.config.mjs
import { defineConfig } from "wmr";

export default defineConfig({
  middleware: [
    (req, res, next) => {
      res.setHeader("Access-Control-Allow-Origin", "*");
      next();
    }
  ]
});
```

**Reply by wilberforce** [reply]
Thank you!
Would you like me to do a pull request to document this here [https://wmr.dev/docs/plugin-api](https://wmr.dev/docs/plugin-api)?

**Reply by marvinhagemeister** [reply]
That would be lovely!

**Reply by wilberforce** [reply]
https://github.com/preactjs/wmr/pull/890
